graylog2-server/src/test/java/org/graylog2/indexer/fieldtypes/IndexFieldTypePollerPeriodicalTest.java

```diff
@@ -62,14 +81,55 @@ void setUp() throws Exception {
                 serverStatus,
                 Duration.minutes(5),
                 scheduler);
+        when(serverStatus.getLifecycle()).thenReturn(Lifecycle.RUNNING);
+        when(cluster.isConnected()).thenReturn(true);
     }
 
     @Test
-    void scheduledExecutionIsSkippedWhenServerIsNotRunning() throws InterruptedException {
+    void scheduledExecutionIsSkippedWhenServerIsNotRunning() {
         when(serverStatus.getLifecycle()).thenReturn(Lifecycle.HALTING);
 
         periodical.doRun();
 
         verifyNoInteractions(cluster);
     }
+
+    @Test
+    void noConcurrentPollingForFieldTypes() {
+        final IndexSetConfig indexSet = IndexSetConfig.builder()
+                .id("indexSet1")
+                .title("Test Index Set")
+                .indexPrefix("test")
+                .shards(2)
+                .creationDate(ZonedDateTime.now())
+                .indexAnalyzer("standard")
+                .indexTemplateName("test")
+                .indexOptimizationMaxNumSegments(2048)
+                .indexOptimizationDisabled(false)
+                .fieldTypeRefreshInterval(org.joda.time.Duration.standardSeconds(1))
+                .retentionStrategy(NoopRetentionStrategyConfig.createDefault())
+                .replicas(1)
+                .rotationStrategy(MessageCountRotationStrategyConfig.createDefault())
+                .build();
+        final List<IndexSetConfig> indexSets = List.of(indexSet);
+        when(indexSetService.findAll()).thenReturn(indexSets);
+
+        final MongoIndexSet mongoIndexSet = mock(MongoIndexSet.class);
+        when(mongoIndexSet.getActiveWriteIndex()).thenReturn("test_0");
+        when(mongoIndexSetFactory.create(eq(indexSet))).thenReturn(mongoIndexSet);
+
+        final ReentrantLock lock = new ReentrantLock();
+        lock.lock();
+        when(indexFieldTypePoller.pollIndex(anyString(), anyString()))
+                .thenAnswer((Answer<Optional<IndexFieldTypesDTO>>) invocationOnMock -> {
+                    lock.lock();
+                    return Optional.empty();
+                });
+
+        periodical.doRun();
+        periodical.doRun();
```

Comment:
Those 2 invocations does not seem to be run in different threads, they are run sequentially in the test's thread.
Therefore, in my opinion, lock usage has completely no impact on the test behavior.

Reply:
Good catch! I overlooked that we are using a single thread executor. Fixed that!